onadata/apps/api/viewsets/xform_viewset.py

```diff
@@ -705,9 +714,23 @@ def csv_import(self, request, *args, **kwargs):
             # `onadata.apps.logger.models.xform.set_object_permissions()`.
             # For safety and clarity, this endpoint now explicitly denies
             # access to all non-owners.
-            raise PermissionDenied
+            raise exceptions.PermissionDenied
         resp = submit_csv(request, xform, request.FILES.get('csv_file'))
         return Response(
             data=resp,
             status=status.HTTP_200_OK if resp.get('error') is None else
             status.HTTP_400_BAD_REQUEST)
+
+    def perform_destroy(self, instance):
+        username = instance.user.username
+        xform_uuid = instance.uuid
+        xform_id_string = instance.id_string
+
+        instance.delete()
+        # Clean up storage
+        default_storage.delete(str(instance.xls))
+        rmdir(f'{username}/form-media/{xform_uuid}')
+        rmdir(f'{username}/exports/{xform_id_string}')
+        # Attachments should have been already deleted. Let's remove the parent
+        # folder entirely anyway in case orphans are still present.
+        rmdir(f'{username}/attachments/{xform_uuid}')
```

Comment:
maybe excessively paranoid, but can we add a check to make sure the UUID and `id_string` are not empty before calling `rmdir()`?